### PR TITLE
Hardening/1127

### LINKED ISSUE #1127: Functest and v1 for op=append for POST  /v2/entities/<entityId>  [+  replace]
More functests for "POST /v2/entities/<entityId>, to add to the PR  #1124

    POST /v2/entities/E1 atribute 'attr2' (and check it is there)
    POST /v2/entities/E1?op=append atribute 'attr3' (and check it is there)
    POST /v2/entities/E1?op=append atribute 'attr3+attr4' (attr3 already exists - make sure attr4 is created and that the operation returns an error about attr3 

Also, if we want to support "appendonly" in v1, tests for that (and possibly modifications in the source code) will be necessary. And, of course, the documentation must be updated to reflect the modifications. Also, changes_next_release will need a line about this (perhaps we should create an issue for only the v1-part-about-append-only).

**Effort: 3 man day**

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #1127

### CODE REVIEW

File: doc/manuals/user/walkthrough_apiv1.md [hunk 1]
```diff
@@ -127,13 +127,13 @@ procedure](../admin/database_admin.md#delete-complete-database).
 To start the broker (as root or using the sudo command):
 
 ```
-/etc/init.d/contextBroker start
+/etc/init.d/contextBroker start
```
Comment (fgalan) [inline doc/manuals/user/walkthrough_apiv1.md]: Don't know why github marks this as change... I haven't (consciously) touched this part.

NTC

File: test/functionalTest/cases/0864_filter_string_scope/filter_string_scope.test [hunk 2]
```diff
@@ -218,4 +218,4 @@ Date: REGEX(.*)
 
 --TEARDOWN--
 brokerStop CB
-#dbDrop CB
+dbDrop CB
```
Comment (fgalan) [inline test/functionalTest/cases/0864_filter_string_scope/filter_string_scope.test]: Not really part of this PR, but spoted and fixed in the process. NTC

File: doc/manuals/user/walkthrough_apiv1.md [hunk 3]
```diff
@@ -886,6 +886,12 @@ Apart from simple values (i.e. strings) for attribute values, you can
 also use complex structures. This is an advance topic, described in
 [this section](structured_attribute_valued.md#structured-attribute-values).
 
+Finally, you can use REPLACE as updateAction. In that case, the entity
+attributes are replaced by the ones in the request. For example, if your
+entity have the atributes A and B and you send an updateContext REPLACE
```
Comment (kzangeli) [inline doc/manuals/user/walkthrough_apiv1.md]: have -> has

File: src/lib/ngsi/UpdateActionType.cpp [hunk 4]
```diff
@@ -47,9 +47,11 @@ std::string UpdateActionType::check
...
 )
 {
-  if ((string == "update") || (string == "UPDATE") || (string == "Update") ||
-      (string == "append") || (string == "APPEND") || (string == "Append") ||
-      (string == "delete") || (string == "DELETE") || (string == "Delete"))
+  if ((string == "update")        || (string == "UPDATE")        || (string == "Update") ||
+      (string == "append")        || (string == "APPEND")        || (string == "Append") ||
+      (string == "append_strict") || (string == "APPEND_STRICT") || (string == "Append_Strict") ||
+      (string == "delete")        || (string == "DELETE")        || (string == "Delete") ||
+      (string == "replace")       || (string == "REPLACE")       || (string == "Replace"))
```
Comment (kzangeli) [inline src/lib/ngsi/UpdateActionType.cpp]: Align ||'s

Comment (fgalan) [inline src/lib/ngsi/UpdateActionType.cpp]: You mean the ones at the end of each line? (the others are aligned)

Comment (kzangeli) [inline src/lib/ngsi/UpdateActionType.cpp]: Yes :-)

File: doc/manuals/user/walkthrough_apiv1.md (hunk 3 above)
Comment (fgalan) [inline doc/manuals/user/walkthrough_apiv1.md]: Fixed in 83655d2

File: src/lib/ngsi/UpdateActionType.cpp (hunk 4 above)
Comment (fgalan) [inline src/lib/ngsi/UpdateActionType.cpp]: Fixed in 83655d2

**Comment by fgalan** [comment]
Please, don't merge after LGTM. Once you put the LGTM, I'll squash + rebase the PR to make it mergeable into develop and let you know when it gets ready for merging.

**Comment by kzangeli** [comment]
LGTM, once synched with develop

**Comment by fgalan** [comment]
Squahsed + rebased. Ready to merge!
